src/celeritas/ext/detail/CelerOpticalPhysics.cc

```diff
@@ -239,13 +279,13 @@ void CelerOpticalPhysics::ConstructProcess()
 
     // NB: boundary is also used later on in loop over particles,
     // though it's only ever applicable to G4OpticalPhotons
-    auto boundary = std::make_unique<G4OpBoundaryProcess>();
+    auto boundary = UniquePtrOwner{std::make_unique<G4OpBoundaryProcess>()};
```

Comment:
Instead of the `UniquePtrOwner` which has unusual semantics, what about:
```suggestion
    auto boundary_storage = std::make_unique<G4OpBoundaryProcess>();
    auto* boundary = boundary_storage.get();
```
and then `boundary.get()` becomes `boundary.release()`?

Or use my updated class suggestion of the class which is somewhere between our two suggestions.